src/inference/src/dev/core_impl.cpp

```diff
@@ -592,6 +592,14 @@ ov::Plugin ov::CoreImpl::get_plugin(const std::string& pluginName) const {
             so = ov::util::load_shared_object(desc.libraryLocation.c_str());
             std::shared_ptr<ov::IPlugin> plugin_impl;
             reinterpret_cast<ov::CreatePluginFunc*>(ov::util::get_symbol(so, ov::create_plugin_function))(plugin_impl);
+            auto plugin_name = plugin_impl->get_device_name();
+
+            // Check that device plugin name is the same as requested
+            // skip for meta
+            // TODO: Create API to detect meta plugins
+            if (plugin_name != "AUTO" && plugin_name != "MULTI") {
```

Comment:
Maybe this [function](https://github.com/openvinotoolkit/openvino/blob/a349dc82f9ac3bca06db3d052ecf37ab1de8b042/src/inference/src/dev/core_impl.cpp#L67) should be in dev API header to detect real/virtual device. This check can be not enough.